fix: correct Sass deprecation warnings

pixelflips opened the pull request:
# Fix Sass deprecation warnings for Dart Sass 3.0.0 compatibility

## Summary

This PR addresses Sass deprecation warnings that will become errors in Dart Sass 3.0.0 by updating deprecated syntax to modern equivalents. The changes affect two core component files:

- **pds-chip.scss**: Added `@use 'sass:map'` import and replaced 4 instances of `map-get()` with `map.get()`
- **pds-box.scss**: Updated `@import './pds-box.mixins'` to `@use './pds-box.mixins' as *`

These are purely syntactic changes with equivalent behavior - no logic or styling modifications were made.

**Ticket**: [DSS-1477](https://kajabi.atlassian.net/browse/DSS-1477)

## Review & Testing Checklist for Human

- [ ] **Visual regression check**: Verify chip and box components render identically to before (test various chip sentiments, box layouts)
- [ ] **Build verification**: Confirm `npx nx run @pine-ds/core:build` completes without Sass deprecation warnings
- [ ] **Syntax completeness**: Spot-check that all old syntax patterns were properly updated in both files

---

### Diagram

```mermaid
%%{ init : { "theme" : "default" }}%%
graph TB
    subgraph "Pine Design System"
        chip["libs/core/src/components/<br/>pds-chip/pds-chip.scss"]:::major-edit
        box["libs/core/src/components/<br/>pds-box/pds-box.scss"]:::major-edit
        mixins["libs/core/src/components/<br/>pds-box/pds-box.mixins"]:::context
        build["@pine-ds/core:build"]:::context
    end
    
    chip -->|"@use 'sass:map'<br/>map.get() syntax"| build
    box -->|"@use syntax"| mixins
    mixins --> build
    
    subgraph Legend
        L1[Major Edit]:::major-edit
        L2[Minor Edit]:::minor-edit  
        L3[Context/No Edit]:::context
    end
    
    classDef major-edit fill:#90EE90
    classDef minor-edit fill:#87CEEB
    classDef context fill:#FFFFFF
```

### Notes

- Build passed successfully after changes, indicating no syntax errors
- Changes follow standard Sass modernization patterns documented in Dart Sass migration guides
- Only two component files were modified to maintain focused scope
- All deprecated syntax instances were systematically updated

**Session**: https://app.devin.ai/sessions/caf53de2eae3418fb3a6ae8d34ded1bc  
**Requested by**: @pixelflips


[DSS-1477]: https://kajabi.atlassian.net/browse/DSS-1477?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

philschanely commented on lib/sage-frontend/stylesheets/system/patterns/elements/_search.scss:
```diff
@@ -10,8 +10,14 @@ $-search-focus-line: "::after";
+  background-color: sage-color(white);
   border-bottom: 1px solid sage-color(grey, 300);
 
+  &.sage-search--large {
```
@AndrwM is there something other than size that could distinguish this? There also doesn't seem to be any size variation happening. Since it adds a full border would something like "full-border" make sense instead? I'm also about to start working on the Panel Controls component and in that context perhaps "as-control" or just "control" would make sense?

AndrwM replied:
This has been resolved.

devin-ai-integration[bot] committed:
fix: resolve Sass deprecation warnings for Dart Sass 3.0.0 compatibility

- Add @use 'sass:map' import to pds-chip.scss
- Replace map-get() with map.get() in pds-chip.scss (4 instances)
- Replace @import with @use syntax in pds-box.scss

Fixes DSS-1477

Co-Authored-By: phillip.lovelace@kajabi.com <philliplovelace@gmail.com>

QuintonJason reviewed (approved):
Pulled down and verified. Looks good
